inference_engine/src/controller.py

```diff
@@ -36,20 +37,30 @@ def _add_io_path(exam_path: str, io_path: str):
     @staticmethod
     def _mark_detection(file_path: str, file_name: str, output_path: str):
         image = Storage.get_exam_image(file_path, file_name)
-        answer_input, index_input = Preprocessing().process(image)
-        index_result = IndexModel(Config.paths.index_model_path).inference(index_input)
-        logging.info(f"Detected index: {index_result}")
-        answer_result = AnswerModel(Config.paths.answer_model_path).inference(answer_input)
+        fields_images = Preprocessing(image).process()
+        ocr_model = OCRModel(Config.paths.index_model_path)
+
+        ocr_results = {f: ocr_model.inference(fields_images[f][0], True if f == Fields.student_id else False)
```

Comment:
I don't quite get what fields exactly are extracted here but overall I understand that they are other fields on the answer sheet than answers.
We plan to move from OCR to box mark detection for student index so I'd rename this to be more universal
```suggestion
        answer_sheet_data = {f: ocr_model.inference(fields_images[f][0], True if f == Fields.student_id else False)
```